lib/core/src/receive_swap.rs

```diff
@@ -270,6 +301,78 @@ impl ReceiveSwapStateHandler {
             (_, Failed) => Ok(()),
         }
     }
+
+    async fn verify_lockup_tx(
```

Comment:
Does this only check if the lockup was broadcasted, or if it's also confirmed?

It's used in both cases in this PR, when handling:
- `RevSwapStates::TransactionMempool`
- `RevSwapStates::TransactionConfirmed`

Maybe it could return `Result<bool>` to cover both cases, based on `lockup_tx_history.is_none()` / `.is_some()`?

Reply:
The `lockup_tx_history` can be confirmed and unconfirmed. `lockup_tx_history.height > 0` will be a confirmed tx.

Reply:
I added verification for the confirmed case to actually verify the tx is confirmed as the swapper reports.